relentless/core.py

```diff
@@ -40,6 +49,13 @@ class Interpolator:
 
     """
     def __init__(self, x, y):
+        x = np.atleast_1d(x)
+        if isinstance(x, (float,int)) or isinstance(y, (float,int)):
+            raise ValueError('x and y must not be scalars')
+        if isinstance(x[0], (tuple,list,np.ndarray)) or isinstance(y[0], (tuple,list,np.ndarray)):
+            raise ValueError('x and y must be 1-dimensional')
+        if not np.all(np.diff(x) > 0):
```

Comment:
You should use something like the other version that I suggested:
```py
if not np.all(x[1:] > x[:-1]):
    raise
```
Your version needs to perform an arithmetic difference between elements, then test that the result is positive. My version performs only a logical comparison between elements. Hence, it's probably about 2x faster if you had a large array.

You have already guaranteed that x has at least 2 elements, so this slice will not fail.